app/uk/gov/hmrc/apiplatformadminapi/services/ApplicationsService.scala

```diff
@@ -22,13 +22,19 @@ import scala.concurrent.{ExecutionContext, Future}
 import uk.gov.hmrc.http.HeaderCarrier
 
 import uk.gov.hmrc.apiplatform.modules.common.domain.models.ApplicationId
+import uk.gov.hmrc.apiplatform.modules.common.services.EitherTHelper
 import uk.gov.hmrc.apiplatformadminapi.connectors.ThirdPartyOrchestratorConnector
-import uk.gov.hmrc.apiplatformadminapi.models.FetchedApplication
+import uk.gov.hmrc.apiplatformadminapi.models.ApplicationWithUsers
 
 @Singleton
-class ApplicationsService @Inject() (thirdPartyOrchestratorConnector: ThirdPartyOrchestratorConnector)(implicit val ec: ExecutionContext) {
+class ApplicationsService @Inject() (thirdPartyOrchestratorConnector: ThirdPartyOrchestratorConnector)(implicit val ec: ExecutionContext)
+    extends EitherTHelper[String] {
```

Comment:
Not sure about this..

Reply:
val E = EitherTHelper.make[String]
  
  def getApplicationWithUsers(applicationId: ApplicationId)(implicit hc: HeaderCarrier): Future[Either[String, ApplicationWithUsers]] = {
    (for {
      applicationResponse <- E.fromOptionF(thirdPartyOrchestratorConnector.getApplication(applicationId), "Application could not be found")
      developers          <- E.liftF(thirdPartyOrchestratorConnector.getApplicationDevelopers(applicationId))
    } yield ApplicationWithUsers.from(applicationResponse, developers))
      .value
  }

Reply:
Happy to implement this a different way with some advice. What I'm trying to achieve is that the call to `getApplicationDevelopers` doesn't happen if the call to `getApplication` returns `None`.